docs/querying/query-context-reference.md

```diff
@@ -44,6 +44,7 @@ Unless otherwise noted, the following parameters apply to all query types, and t
 |Parameter          |Default                                 | Description          |
 |-------------------|----------------------------------------|----------------------|
 |`timeout`          | `druid.server.http.defaultQueryTimeout`| Query timeout in millis, beyond which unfinished queries will be cancelled. 0 timeout means `no timeout` (up to the server-side maximum query timeout, `druid.server.http.maxQueryTimeout`). To set the default timeout and maximum timeout, see [Broker configuration](../configuration/index.md#broker) |
+|`perSegmentTimeout`| `null`                                 | Per-segment processing timeout in millis, beyond which unfinished queries will be cancelled. Should be ≤ `timeout`. 0 `perSegmentTimeout` means `no per-segment timeout`. Generally, a standard default should be O(X seconds). A cluster-wide default value for this query context can be specified via `druid.query.default.context.perSegmentTimeout`.|
```

Comment:
```suggestion
|`perSegmentTimeout`| `null`                                 | Per-segment processing timeout in millis, beyond which unfinished queries occupying the processing pool will be cancelled. Should be ≤ `timeout`. Setting `perSegmentTimeout` to 0 means there's no per-segment timeout (up to the server-side maximum per segment timeout `druid.query.default.context.perSegmentTimeout`). Generally, a standard default should be in the order of a few seconds (e.g., 5 seconds). A cluster-wide default value for this query context can be specified via `druid.query.default.context.perSegmentTimeout`.|
```

Reply:
I think using -1 as the default to disable per segment timeouts would be clearer. However, it's consistent with `timeout`'s default strategy, so this is okay